Adds type checks and float casting in distaz fixes #2497

heavelock opened the pull request:
Casts non-native floats to native ones. Adds type tests to that method and the next one.
fixes #2497

megies commented:
+TESTS:clients.iris

megies commented:
Restarted Travis so teh client tests are run.

heavelock commented:
My tests pass. 
The failing test thing is one of evalresp's. Only on linux, the `fap` file has one space between values instead of expected two. The writing method is from 2011/2013 by Krischer and Barsch, the evalresp fetcher is from 2011 by Barsch. Thus it seems that it's rather something system dependent, not the code itself.

megies commented:
Lets get some fresh CI, looked like some doctests need fixing or maybe its fixed in master, we'll see

heavelock commented:
So it looks like IRIS updated on their side entries that the tests suite is checking against because returned values changes. On our side the values were also changed but only in the test data file, not in doctests. But even there there is one value that changed once again. 

Should I fix those comparisons and data stored in doctests and test data file? 

The evalresp error stayed the same as before, either fetched data chenged on writing to ASCII broke.

megies commented:
> Should I fix data stored in doctests and test data file?

Yeah please adapt them if you don't mind (even if they might be unrelated I guess). You can use ellipsis in doctests `...` (need to also put `# doctest: +ELLIPSIS`) to make them more forgiving wherever it seems fragile.

heavelock commented:
I changed the expected values. In the end for the sacpz affected was only last digit of lat and lon which might suggest that there is a rounding error somewhere. 
I adjusted also evalresp. I ended up basically adding spaces between values where there was single one and removing where was a double one. Weird.

I will check what Travis will say and see if I should dig deeper.

megies commented on obspy/clients/iris/tests/test_client.py:
```diff
@@ -166,7 +166,7 @@ def test_evalresp(self):
             with open(tempfile, 'rt') as fp:
                 self.assertEqual(fp.readline(),
-                                 '1.000000E-05  1.055999E+04  1.792007E+02\n')
+                                 '1.000000E-05 1.055999E+04 1.792007E+02\n')
```
if these keep making problems you could also `.split()` and compare as a list

heavelock replied:
It looks like it's going to pass so I would leave it as it is.

heavelock replied:
It looks like it should be okay, but yeah, that would be a better approach.

megies replied:
I don't care.. if it's passes for now I'm happy

heavelock commented:
Rebased and force pushed, should ready to merge after CI.

heavelock commented:
Hmm, after that rebase, at 2.7 type assertion for `str` started to complain. The reason is that the parsed string is returned as a `str` but I'm comparing it to py2-3 compatible string `future.types.newstr.BaseNewStr`. Casting with use of `str()` fixes it. But do you have any idea why it happened now? It's weird.

megies commented:
> Hmm, after that rebase, at 2.7 type assertion for `str` started to complain. The reason is that the parsed string is returned as a `str` but I'm comparing it to py2-3 compatible string `future.types.newstr.BaseNewStr`. Casting with use of `str()` fixes it. But do you have any idea why it happened now? It's weird.

We had stuff like that happen before, but I can't be bothered to look it up right now and after 1.2.0 is released we're getting rid of all that anyway.

megies commented:
Thanks @heavelock
